internal/service/ec2/ec2_instance_metadata_defaults.go

```diff
@@ -5,219 +5,206 @@ package ec2
-			err.Error(),
-		)
+	conn := r.Meta().EC2Client(ctx)
+
+	input := &ec2.ModifyInstanceMetadataDefaultsInput{}
+	response.Diagnostics.Append(fwflex.Expand(ctx, data, input)...)
+	if response.Diagnostics.HasError() {
 		return
 	}
 
-	// Convert what we read from the AWS API to our format
-	state = r.instanceMetadataDefaultsToPlan(out)
+	_, err := conn.ModifyInstanceMetadataDefaults(ctx, input)
 
-	resp.Diagnostics.Append(resp.State.Set(ctx, &state)...)
-}
+	if err != nil {
+		response.Diagnostics.AddError("creating EC2 Instance Metadata Defaults", err.Error())
 
-func (r *resourceEC2InstanceMetadataDefaults) Create(ctx context.Context, req resource.CreateRequest, resp *resource.CreateResponse) {
-	var plan resourceEC2InstanceMetadataDefaultsData
-	resp.Diagnostics.Append(req.Plan.Get(ctx, &plan)...)
-	if resp.Diagnostics.HasError() {
 		return
 	}
 
-	in := r.planToModifyInstanceMetadataDefaultsInput(&plan)
-	if err := r.updateDefaultInstanceMetadataDefaults(ctx, in, create.ErrActionCreating); err != nil {
-		resp.Diagnostics.AddError(err.Error(), "")
-	}
-
-	resp.Diagnostics.Append(resp.State.Set(ctx, plan)...)
+	response.Diagnostics.Append(response.State.Set(ctx, data)...)
 }
 
-// Update is very similar to Create as AWS has a single API call ModifyInstanceMetadataDefaults
-func (r *resourceEC2InstanceMetadataDefaults) Update(ctx context.Context, req resource.UpdateRequest, resp *resource.UpdateResponse) {
-	var plan, state resourceEC2InstanceMetadataDefaultsData
-	resp.Diagnostics.Append(req.Plan.Get(ctx, &plan)...)
-	resp.Diagnostics.Append(req.State.Get(ctx, &state)...)
-
-	in := r.planToModifyInstanceMetadataDefaultsInput(&plan)
-	if err := r.updateDefaultInstanceMetadataDefaults(ctx, in, create.ErrActionUpdating); err != nil {
-		resp.Diagnostics.AddError(err.Error(), "")
+func (r *instanceMetadataDefaultsResource) Read(ctx context.Context, request resource.ReadRequest, response *resource.ReadResponse) {
+	var data instanceMetadataDefaultsModel
+	response.Diagnostics.Append(request.State.Get(ctx, &data)...)
+	if response.Diagnostics.HasError() {
+		return
 	}
 
-	resp.Diagnostics.Append(resp.State.Set(ctx, &plan)...)
-}
+	conn := r.Meta().EC2Client(ctx)
 
-func (r *resourceEC2InstanceMetadataDefaults) Delete(ctx context.Context, req resource.DeleteRequest, resp *resource.DeleteResponse) {
-	var state resourceEC2InstanceMetadataDefaultsData
-	resp.Diagnostics.Append(req.State.Get(ctx, &state)...)
+	output, err := findInstanceMetadataDefaults(ctx, conn)
 
-	in := &ec2.ModifyInstanceMetadataDefaultsInput{
-		HttpEndpoint:            awstypes.DefaultInstanceMetadataEndpointStateNoPreference,
-		HttpPutResponseHopLimit: aws.Int32(-1), // -1 means "no preference"
-		HttpTokens:              awstypes.MetadataDefaultHttpTokensStateNoPreference,
-		InstanceMetadataTags:    awstypes.DefaultInstanceMetadataTagsStateNoPreference,
-	}
+	if tfresource.NotFound(err) {
```

Comment:
GetInstanceMetadataDefaults always return something (even if no defaults were explicitely set through ModifyInstanceMetadataDefaults), so I think this check is not necessary FWIW